src/boost_histogram/_internal/hist.py

```diff
@@ -1067,11 +1067,12 @@ def variances(self, flow: bool = False) -> Optional["np.typing.NDArray[Any]"]:
             return view if self._variance_known else None
 
         if hasattr(view, "sum_of_weights"):
+            valid = view.sum_of_weights**2 > view.sum_of_weights_squared  # type: ignore[union-attr]
```

Comment:
Actually, I think this can just be:

```suggestion
            valid = view.sum_of_weights != view.sum_of_weights_squared  # type: ignore[union-attr]
```

This will either both be 0, or both be identical (counts==1 exactly), or this is a valid calculation. Comparing floating point numbers is a little scary, but I think this is fine.

Reply:
Also I need to add a test for this 😳